InsightLogParser.Client/Spider.cs

```diff
@@ -650,10 +662,9 @@ private void WriteWaypointMessage(RouteNode node, int current, int max)
 
         public void NextRouteWaypoint()
         {
-            var current = _puzzleRouter.CurrentNode();
-            var currentCoordinate = current?.Node.Puzzle.PrimaryCoordinate!.Value ?? _teleportManager.GetLastTeleport() ?? default;
```

Comment:
This line was the bug from #1.  The assumption here is that you're at the previous puzzle because you just solved it, but this is not true if you're using the next feature to advance through the route.